syntax = "proto3";
package proto;

message Chat {
	string messageId = 1;
	string roomId = 2;
	string userId = 3;
	int64 timestamp = 4;
	string type = 5;
	string clientUid = 6;
	string data = 7;
}

message key {
	string key = 1;
}

message Empty {}



message Status {
	bool ok = 1;
}

service Backup {
    rpc OnMessageIn(Chat) returns (Empty);
	rpc IsReady(Empty) returns (Status);
	rpc GetKey(Empty) returns (key);
	rpc EncryptedMessage(Chat) returns (Chat);
	rpc DecryptedMessage(Chat) returns (Chat);
}
